setup.py

```diff
@@ -1,15 +1,20 @@
 import os
 import subprocess
 import warnings
-from packaging import version
-import pip
+from setuptools._vendor.packaging import version
```

Comment:
Like already discussed: I do not really like using `setuptools._vendor.packaging` and would prefer `packaging`. https://github.com/precice/python-bindings/pull/63#issuecomment-717839037 is, of course, an argument to patch this on the spack-side, if necessary. I'm a bit surprised that this does not work. Any idea?

Reply:
I would assume that this is either a Spack issue or a user (my) issue using Spack. If I use the branch where the `packaging` module is installed, I can access `packaging` in an interactive python session, but the Spack recipe does not find it. When I set up the recipe I simply gave up at some point and went for the crude fix. (I would prefer a proper fix though.)

Reply:
I installed `py-packaging` manually in my spack env and observed the same issue using [your branch](https://github.com/ajaust/spack/tree/version-from-packaging):
```
==> Installing py-pyprecice
==> No binary for py-pyprecice found: installing from source
==> Using cached archive: /home/benjamin/spack/var/spack/cache/_source-cache/archive/97/972f574549344b6155a8dd415b6d82512e00fa154ca25ae7e36b68d4d2ed2cf4.tar.gz
==> py-pyprecice: Executing phase: 'build_ext'
==> Error: ProcessError: Command exited with status 1:
    '/home/benjamin/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/python-3.8.6-wlcjv5qbqqiqwfkkkksxjyh2k2ru2hvw/bin/python3.8' '-s' 'setup.py' '--no-user-cfg' 'build_ext' '--include-dirs=/home/benjamin/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/precice-2.1.1-utdkrqewi3yphbgst2itc7ak4l2puu2w/include' '--library-dirs=/home/benjamin/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/precice-2.1.1-utdkrqewi3yphbgst2itc7ak4l2puu2w/lib'
See build log for details:
  /tmp/benjamin/spack-stage/spack-stage-py-pyprecice-2.1.1.1-ummpnvtonm5vajataudie3yx2vu3vnjc/spack-build-out.txt

benjamin@benjamin-ThinkPad-X1-Yoga-2nd:~/spack$ cat /tmp/benjamin/spack-stage/spack-stage-py-pyprecice-2.1.1.1-ummpnvtonm5vajataudie3yx2vu3vnjc/spack-build-out.txt 
==> py-pyprecice: Executing phase: 'build_ext'
==> [2020-10-28-12:11:27.387199] '/home/benjamin/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/python-3.8.6-wlcjv5qbqqiqwfkkkksxjyh2k2ru2hvw/bin/python3.8' '-s' 'setup.py' '--no-user-cfg' 'build_ext' '--include-dirs=/home/benjamin/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/precice-2.1.1-utdkrqewi3yphbgst2itc7ak4l2puu2w/include' '--library-dirs=/home/benjamin/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/precice-2.1.1-utdkrqewi3yphbgst2itc7ak4l2puu2w/lib'
Traceback (most recent call last):
  File "setup.py", line 4, in <module>
    from packaging import version
ModuleNotFoundError: No module named 'packaging'
benjamin@benjamin-ThinkPad-X1-Yoga-2nd:~/spack$ python3 -c "import packaging; print(packaging)"
<module 'packaging' from '/home/benjamin/spack/var/spack/environments/myproject/.spack-env/view/lib/python3.8/site-packages/packaging-19.2-py3.8.egg/packaging/__init__.py'>
```

Reply:
If you use my branch, it should install `py-packaging` anyway since I specify it as a dependency for `py-pyprecice`.

I added a remark in the Spack PR. Maybe they know more.

Reply:
See https://github.com/precice/python-bindings/pull/63#discussion_r513398057. I think we should try to solve this issue either on the spack-side or in another PR. Ok? Then we can "resolve" this conversation.

Reply:
Fixed in 5b35c3c.